subsys/usb/device_next/usbd_device.c

```diff
@@ -165,7 +166,19 @@ int usbd_wakeup_request(struct usbd_contex *const uds_ctx)
 		goto wakeup_request_error;
 	}
 
+	k_sem_reset(&uds_ctx->rwup_sem);
```

Comment:
Isn't this prone to race conditions where the host resumes the device before `usbd_wakeup_request()` obtains the usbd lock?

Using `k_event` that would have a bit that signifies `not suspended` would be immune to such race. Then the event would be set and cleared in `usbd_status_suspended()` with `k_event_set_masked(&uds_ctx->events, value ? 0 : UDS_NOT_SUSPENDED_BIT, UDS_NOT_SUSPENDED_BIT);` and the wait here would be changed to `k_event_wait(&uds_ctx->events, UDS_NOT_SUSPENDED_BIT, false, resume_timeout);`.

The name `not suspended` might not be the best, an alternative could be `active` but it might not be that good either. It would be important to make it `not suspended` because `k_event_wait` only allows to wait until an event bit gets set (there is no wait function for event bit to become cleared).